# Chapter 5 Ex 
Algorithm Workbench 1, 2 
Programming Exercises 1, 17 ('Prime Numbers), 18 (prime number list), 19 

# Algorithm Workbench 1
Write a function named time_ten, the function should accept an argument and display the product of its argument multiplied times 10.

In [3]:
# Define a function named times_ten that takes one argument.
def times_ten(number):
    # Multiply the number by 10 and return the result.
    return number * 10

product = times_ten(5)
print(f"The product of 5 multiplied by 10 is: {product}")

The product of 5 multiplied by 10 is: 50


Explannation: This code defines a 'mini-program' inside your main program, which we call a function, that does a specific job: it takes any number you give it, makes it ten times bigger, and then shows you this new, bigger number. The name of this mini-program is times_ten. It's like telling your computer, "Whenever I say 'times_ten' and give you a number, please multiply that number by 10 and tell me what you get."

# Algorithm Workbench 2
Examine the following function header, then write a statement that calls the function, passing 12 as an argument.
def show_value(quantity): 

In [9]:
# This is the function as provided.
def show_value(quantity):
    # This function now prints the value of 'quantity'.
    print(f"The value is: {quantity}")
show_value(12)

The value is: 12


Explannnation: When you run this code, it will display the message "The value is: 12" in the console. The function show_value is being called with the number 12, and it uses the print function to show that number to you.

# Programming Exercise 1
Write a program that asks the user to enter a distance in kilometers then converts that distance to miles, use formula : Miles = Kilometers x 0.6214

In [1]:
# Ask the user to enter a distance in kilometers.
kilometers = float(input("Enter distance in kilometers: "))

# Convert the distance to miles using the conversion formula.
miles = kilometers * 0.6214

# Display the distance in miles.
print(f"The distance in miles is: {miles}")

Enter distance in kilometers: 62
The distance in miles is: 38.526799999999994


Explannation: The program starts by asking you to type in a distance in kilometers. After you enter the distance, it takes that number and multiplies it by 0.6214 because that's how you change kilometers into miles. Then, it shows you the result in miles. It's a handy way to figure out how far something is in miles if you only

# Programming Exercise 17
Prime numbers, write a Boolean function named is_prime which takes an integer as an argument and returns trues if the argument is a prime number, or false.display if the number is prime

In [2]:
# Define a function to check if a number is prime.
def is_prime(number):
    # If the number is less than 2, it is not prime.
    if number < 2:
        return False
    # Check for factors of number from 2 to the square root of the number.
    for i in range(2, int(number**0.5) + 1):
        # If number is divisible by i, then it is not prime.
        if number % i == 0:
            return False
    # If no factors are found, the number is prime.
    return True

# Ask the user to enter a number.
user_number = int(input("Enter a number to check if it's prime: "))

# Use the is_prime function to check if the number is prime.
if is_prime(user_number):
    print(f"{user_number} is a prime number.")
else:
    print(f"{user_number} is not a prime number.")


Enter a number to check if it's prime: 199
199 is a prime number.


Explannation: The program defines a special task named is_prime that can figure out if the number you give it is a prime number or not. A prime number is one that you can't divide evenly by any other number except 1 and itself. The program then asks you to give it a number to check. Once you type in a number, it uses the is_prime task to see if your number is prime. After it finishes checking, it will tell you whether your number is prime or not. If it is prime, it'll say "199 is a prime number." If not, it'll say "199 is not a prime number."

# Programming Exercise 18 
Prime number list, write another program that displays all the prime numbers from 1 to 100, the program should have a loop that calls it_prime function

In [3]:
# Use the is_prime function from the previous example.
def is_prime(number):
    if number < 2:
        return False
    for i in range(2, int(number**0.5) + 1):
        if number % i == 0:
            return False
    return True

# Create a loop to check each number from 1 to 100.
for number in range(1, 101):  # Starts at 1 and goes up to 100
    # Call the is_prime function to check if the number is prime.
    if is_prime(number):
        # If the number is prime, print it.
        print(number, "is a prime number.")

# At the end of the program, print a message.
print("These are all the prime numbers between 1 and 100.")

2 is a prime number.
3 is a prime number.
5 is a prime number.
7 is a prime number.
11 is a prime number.
13 is a prime number.
17 is a prime number.
19 is a prime number.
23 is a prime number.
29 is a prime number.
31 is a prime number.
37 is a prime number.
41 is a prime number.
43 is a prime number.
47 is a prime number.
53 is a prime number.
59 is a prime number.
61 is a prime number.
67 is a prime number.
71 is a prime number.
73 is a prime number.
79 is a prime number.
83 is a prime number.
89 is a prime number.
97 is a prime number.
These are all the prime numbers between 1 and 100.


Explnation: This program uses the is_prime task we defined earlier to check each number from 1 to 100. It goes through each number one by one and asks, "Hey, is_prime, is this number a prime number?" If is_prime says yes, the program prints out the number and tells you it's a prime number. After it checks all the numbers up to 100, it prints out a message saying that these are all the prime numbers between 1 and 100

# Programming Exercuse 19
Suppose you have certain amount of money in savings account that earns compound monthly interest, and write a program to calculate the amount you will have after specific number of month 

In [4]:
# Define a function to calculate the future value of an account with compound interest.
def calculate_future_value(present_value, monthly_interest_rate, months):
    # Calculate the future value using the compound interest formula.
    future_value = present_value * ((1 + monthly_interest_rate) ** months)
    # Return the calculated future value.
    return future_value

# Ask the user for the present value of the account.
present_value = float(input("Enter the present value of the account: "))

# Ask the user for the monthly interest rate (as a decimal).
monthly_interest_rate = float(input("Enter the monthly interest rate (as a decimal): "))

# Ask the user for the number of months.
months = int(input("Enter the number of months: "))

# Call the function to calculate the future value.
future_value = calculate_future_value(present_value, monthly_interest_rate, months)

# Display the future value to the user.
print(f"The future value of the account after {months} months will be: {future_value:.2f}")


Enter the present value of the account: 883554
Enter the monthly interest rate (as a decimal): 0.04
Enter the number of months: 12
The future value of the account after 12 months will be: 1414598.42


Explanation: This program helps you figure out how much money you'll have in the future if you earn interest every month. It asks you for three things: how much money you're starting with, what the monthly interest rate is (like if you get 1% interest, you'd type 0.01), and how many months you're going to leave the money to grow. Then it uses a special math formula to calculate your future money, considering the interest that adds up each month. Finally, it tells you the result, showing you how much money you can expect to have after the given time. 